docs/website/docs/general-usage/http/rest-client.md

```diff
@@ -0,0 +1,413 @@
+
+The `RESTClient.paginate()` method is specifically designed to handle paginated responses, yielding `PageData` instances for each page:
+
+```py
+for page in client.paginate("/posts"):
+    print(page)
+```
+
+If `paginator` is not specified, the `paginate()` method will attempt to automatically detect the pagination mechanism used by the API. If the API uses a standard pagination mechanism like having a `next` link in the response's headers or JSON body, the `paginate()` method will handle this automatically. Otherwise, you can specify a paginator object explicitly or implement a custom paginator.
+
+### PageData
+
+Each `PageData` instance contains the data for a single page, along with context like the original request and response objects, allowing for detailed inspection. The `PageData` is a list-like object that contains the following attributes:
+
+- `request`: The original request object.
+- `response`: The response object.
+- `paginator`: The paginator object used to paginate the response.
+- `auth`: The authentication object used for the request.
+
+### Selecting data from the response
+
+The `data_selector` parameter in the `RESTClient` constructor specifies a JSONPath selector for extracting data from the response.
+
+For example, if the API response looks like this:
+
+```json
+{
+    "posts": [
+        {"id": 1, "title": "Post 1"},
+        {"id": 2, "title": "Post 2"},
+        {"id": 3, "title": "Post 3"}
+    ]
+}
+```
+
+The `data_selector` should be set to `"posts"` to extract the list of posts from the response.
+
+For a nested structure like this:
+
+```json
+{
+    "results": {
+        "posts": [
+            {"id": 1, "title": "Post 1"},
+            {"id": 2, "title": "Post 2"},
+            {"id": 3, "title": "Post 3"}
+        ]
+    }
+}
+```
+
+The `data_selector` should be set to `"results.posts"`. Read more about [JSONPath syntax](https://github.com/h2non/jsonpath-ng?tab=readme-ov-file#jsonpath-syntax).
+
+### Paginators
+
+Paginators are used to handle paginated responses. The `RESTClient` class comes with built-in paginators for common pagination mechanisms:
+
+- [JSONResponsePaginator](#jsonresponsepaginator) - link to the next page is included in the JSON response.
+- [HeaderLinkPaginator](#headerlinkpaginator) - link to the next page is included in the response headers.
+- [OffsetPaginator](#offsetpaginator) - pagination based on offset and limit query parameters.
+- [JSONResponseCursorPaginator](#jsonresponsecursorpaginator) - pagination based on a cursor in the JSON response.
+
+If the API uses a non-standard pagination, you can [implement a custom paginator](#implementing-a-custom-paginator) by subclassing the `BasePaginator` class.
+
+#### JSONResponsePaginator
+
+`JSONResponsePaginator` is designed for APIs where the next page URL is included in the response's JSON body. This paginator uses a JSON path to locate the next page URL within the JSON response.
+
+##### Parameters:
```

Comment:
NIT:
The 5th and 6th headings looks very strange, smaller than a regular text :D 
Maybe just use bold?
![image](https://github.com/dlt-hub/dlt/assets/9139429/eeadb504-5dfb-414b-a047-d7a68c1a5df3)

Reply:
Good point